### DISCUSSION TITLE
Why Flutter used Dart?

### DISCUSSION THREAD

**Discussion by khadkha** [community/community#28061] [Mobile] [answered]
Why Flutter used Dart?

**Comment by PLLara** [comment]
Their first alternative was javascript but it turned out to be excessively slow. Then they picked the same document that javascript is based on ( ECMA ) and made a new language without all the historical flaws that JS has, dart.
They could've other languages like c#, java, c++, but all of them are too much complicated, and flutter is meant to be both performatic and easy to learn, so they literally invented a new programming language just for this pourpose.
Some people would say that kotlin was a good choice, but google is google, we never know exactly why they do what they do.
Anyhow, languages like Js, Dart and Kotlin are very alike, jumping from one to another should be easy to experienced developers.

**Comment by surajadkhari** [comment, marked as answer]
Flutter uses Dart as Dart allows Flutter to avoid the need for a separate declarative layout language like JSX and XML. The layout of Dart is declarative and programmatic, and it makes it easy for developers to read and visualize it very quickly and effortlessly. In addition to that, it makes it easy for Flutter to provide additional tooling as the layout is in one language and commonplace.

Another important reason why the duo is popular is that Dart, if required, uses the Just In Time compilation. This drastically reduces the time of development and responds faster.

Most of the Dart features are similar to that of static and dynamic languages, and it makes it easy for developers to learn and understand Dart easily and quickly.

If you see closely, not all the features mentioned above are extraordinary and superior. However, when these features are combined, they serve as the best implementer for Flutter. It is not an overstatement to say that Flutter app development would not be as smooth and streamlined without Dart for the [Flutter app development company.](https://kodytechnolab.com/flutter-app-development)

**Reply by khadkha** [reply]
Thank you so much

**Reply by PLLara** [reply]
I could be wrong, but i believe the fact they haven't choose a declarative layout language is just a matter of choice and design and not a particular reason for choosing dart over any other object oriented programming language.
Javascript for example has both aproaches with React.createElement and JSX.

**Comment by jaiswal4sudeep** [comment]
Flutter uses Dart as Dart allows Flutter to avoid the need for a separate declarative layout language like JSX and XML. The layout of Dart is declarative and programmatic, and it makes it easy for developers to read and visualize it very quickly and effortlessly.
